### PR TITLE
Replace deprecated function causing gradle sync to fail

### PR DESCRIPTION
**Pull request by YahiaAngelo**

### CODE REVIEW

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -120,7 +120,7 @@ tasks {
         // pluginVersion is based on the SemVer (https://semver.org) and supports pre-release labels, like 2.1.7-alpha.3
         // Specify pre-release label to publish the plugin in a custom Release Channel automatically. Read more:
         // https://plugins.jetbrains.com/docs/intellij/deployment.html#specifying-a-release-channel
-        channels.set(listOf(properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first()))
+        channels.set(listOf(properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.')[0]))
```
Comment (bric3) [inline plugin/build.gradle.kts:123]: **question:** How can it be deprecated ?

https://github.com/JetBrains/kotlin/blob/v1.9.22/libraries/stdlib/common/src/generated/_Arrays.kt#L1005-L1014

Also I have no issue to load this project on IJ 2023.3 which is the most up to date about Gradle and Kotlin. I have no report about this.
image

Comment (YahiaAngelo) [reply plugin/build.gradle.kts:123]: Here's the error I got, using the latest IJ as well:
image

I also noticed that it uses `.first()` function from jdk not kotlin-stdlib like yours.
Maybe IDE settings issue?

Comment (bric3) [reply plugin/build.gradle.kts:123]: Ah interesting, thank you! I forgot that JDK 21 added sequenced collections (which adds `getFirst()`). So You are probably running Gradle with a JDK 21, which explains why there's this warning.

Unfortunately Gradle can be run by earlier JDKs that don't have `getFirst()`. So I'm inclined to keep the kotlin version. However I don't understand why the sync is failing on your end then ?

Comment (bric3) [reply plugin/build.gradle.kts:123]: I could reproduce locally ! Still that feels odd. 
Meanwhile you should try to configure a lower JDK to run Gradle in IJ.

```
e: file:///Users/brice.dutheil/opensource/excalidraw-jetbrains-plugin/plugin/build.gradle.kts:122:18: None of the following functions can be called with the arguments supplied: 
public abstract fun set(elements: (Mutable)Iterable<String!>?): Unit defined in org.gradle.api.provider.ListProperty
public abstract fun set(provider: Provider<out (Mutable)Iterable<String!>!>!): Unit defined in org.gradle.api.provider.ListProperty

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/brice.dutheil/opensource/excalidraw-jetbrains-plugin/plugin/build.gradle.kts' line: 122

* What went wrong:
Script compilation error:

  Line 122:         channels.set(listOf(properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first()))
                             ^ None of the following functions can be called with the arguments supplied: 
                                 public abstract fun set(elements: (Mutable)Iterable<String!>?): Unit defined in org.gradle.api.provider.ListProperty
                                 public abstract fun set(provider: Provider<out (Mutable)Iterable<String!>!>!): Unit defined in org.gradle.api.provider.ListProperty
```

Comment (bric3) [reply plugin/build.gradle.kts:123]: I believe you spot a fun bug there : https://github.com/gradle/gradle/issues/27699

**Comment by bric3** [comment]
At this time I prefer to sit a bit on this, current workaround is to use a JDK < 21 to run Gradle.

image

**Comment by bric3** [comment]
After a night of sleep, I've decided to avoid the issue, this code was a straw man parsing to get the version label. I'll push an update that will use a regular expression instead.

**Pull request closed without merging**